### DISCUSSION TITLE
Share The Love! Tell us about how you use Grapevine!

### DISCUSSION THREAD

**Discussion by scottoffen** [scottoffen/grapevine#13] [Show and tell]
Are you using Grapevine and loving it? We'd love to hear about it!

**Comment by Lroy** [comment]
I have just found Grapevine while searching for something that does what I think Grapevine does.

I need to serve up static VUE.js-based content to a browser, but that content will contain API calls back to the server to fetch specific content from my application.  As far as I can tell at this moment, and I must admit I have only been messing around with this for about an hour, Grapevine looks like it would work for me.

I am wondering what would be involved in making my API "HTTPS://<ipaddr>:xxx/API/whatever".  Is there any reason why Grapevine would NOT be able to work with HTTPS?

Larry

**Reply by Lroy** [reply]
Also,

I have pulled down V4.2.2 from NuGet in my project, and do not seem to have this method:

await context.Response.SendResponseAsync("Successfully hit the test route!")

Should I be thinking about switching up to V5.x?

**Reply by Lroy** [reply]
BTW, I found the docs for V4.2.2, but I'm wondering whether I should spend much time reading that, if I you recommend that I should be adopting V5.  When do believe V5 will be GA?

**Reply by scottoffen** [reply]
Grapevine 5 won't be in GA until I've finished the documentation. This is because as I work through documenting the project, I'm finding inconsistencies that I want to clean up, and somethings this changes the API. I'm targeting late February 2021, and I'm on track for that.

Having said that, it's pretty stable right now. There are docs for both 4 and 5 that can direct you on how to set up SSL. There is almost no code in Grapevine to make that happen, it's all about opening ports and binding certificates on the OS.

**Comment by Lroy** [comment]
I have been poking around for good SSL info… anything you might have would be appreciated!

I’m going to give it another go tomorrow morning.

I am currently trying to figure out whether I need to establish ACL entries for both HTTP and HTTPS… I don’t think I do, but I’m not positive yet.

**Reply by scottoffen** [reply]
The [HTTPS/SSL documentation for GV5](https://scottoffen.github.io/grapevine/docs/tutorials/using-https-ssl) is much better than what exists for GV4.

The only different is the very first step. Instead of the GV5 way of adding prefixes (GV5 supports listening on multiple prefixes):

```csharp
server.Prefixes.Add("https://*:443/");
```

You would use the GV4 way (where there is only one prefix):

```csharp
server.UseHttps = true;
```

Don't forget to [open a port in the windows firewall](https://scottoffen.github.io/grapevine/docs/tutorials/windows-firewall/) and [configure the namespace reservations](https://scottoffen.github.io/grapevine/docs/tutorials/configure-namespace-reservation/).

**Reply by Lroy** [reply]
Scott,

Finally got the ACL/Self-cert/SSL Binding all to work, and everything is going well.  Thanks for your help!

**Comment by sanchezpablo** [comment]
I found quite interesting the project, I needed to embed a simple server as backend for an electron application which will do some database/devices work. The first impression is great, simple to the point. The only concern now is the CPU usage, which is 25% no matter what, even in idle state without requests at all. Can we do something for this?

**Reply by Lroy** [reply]
I don't know about CPU usage for Grapevine V5, but Grapevine V4.2.2 uses zero CPU when idle (listening, but not serving up any content).

**Reply by scottoffen** [reply]
There were some performance issues, they have been resolved and should be in the next release.

**Comment by cbritton** [comment]
I am developing an application that consumes messages from AWS SNS and processes the content.  The application needs to be small and fast and I don't want to manage a full fledged HTTP server since it just has a single post interface and I don't want a heavy library to go with it. Grapevine is perfect for what I need.

**Reply by scottoffen** [reply]
This is the niche I was going for! If this is an open source project, feel free to share a link. Thanks!

**Comment by minecraftchest1** [comment]
I am thinking of using this project as a way to allow one application to trigger an event in another one. The use case that I have is a cross-platform parental control software package that I am working on. I needed a way for Windows to tell the control daemon that a user has signed in, out, locked/unlocked their workstation, etc. This was the best solution that I could find that was cross-platform and simple. 

Thanks for this project. I hope to have source code to share in the future.

**Reply by scottoffen** [reply]
That sounds cool. How is it working out so far?

**Reply by minecraftchest1** [reply]
I haven't worked on it much (too many other projects). However, all of my testing has gone well. Hardest part is designing the api.

**Comment by XaverB** [comment]
The research institute (HEAL) at the university of applied science Hagenberg created an application called [HeuristicLab](https://github.com/heal-research/HeuristicLab).
I am allowed to extend this application by a RestService for easier scripting, plotting and so on.
For this I am using grapevine!

Thanks for your great work.

**Comment by GrLawrence** [comment]
Just found this project looking for something lightweight to embed a REST service within an industrial automation machine. I've been experimenting and I think this fits nicely!  Thanks for all of your work on this project!
